GeneralsMD/Code/GameEngine/Source/GameLogic/Object/Behavior/PoisonedBehavior.cpp

```diff
@@ -158,6 +159,9 @@ void PoisonedBehavior::startPoisonedEffects( const DamageInfo *damageInfo )
 
 	// We are going to take the damage dealt by the original poisoner every so often for a while.
 	m_poisonDamageAmount = damageInfo->out.m_actualDamageDealt;
+#if !RETAIL_COMPATIBLE_CRC
+	m_poisonSource = damageInfo->in.m_sourceID;
```

Comment:
This needs to be configurable in PoisonedBehavior INI, otherwise this is a controversial change that affects all derived products. That is, unless community can already decide that this change is universally desirable.

Reference INI fields are:
```
ExperienceSinkForRider
AwardXPForTriggering
```

When it is opt-in in INI, then we also do not need RETAIL_COMPATIBLE_CRC (but still need RETAIL_COMPATIBLE_XFER_SAVE in Xfer)

Reply:
It does not; the experience I saw gained from that was the actual sprayer weapon `Chem_ToxinTruckSprayerGamma`, which I realised upon further testing was dealing (very minor) damage in addition to the puddle.